packages/shared_preferences/shared_preferences/test/shared_preferences_async_test.dart

```diff
@@ -0,0 +1,617 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences/shared_preference_async.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_async_platform_interface.dart';
+import 'package:shared_preferences_platform_interface/types.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  const String stringKey = 'testString';
+  const String boolKey = 'testBool';
+  const String intKey = 'testInt';
+  const String doubleKey = 'testDouble';
+  const String listKey = 'testList';
+
+  const String testString = 'hello world';
+  const bool testBool = true;
+  const int testInt = 42;
+  const double testDouble = 3.14159;
+  const List<String> testList = <String>['foo', 'bar'];
+
+  group('Async', () {
+    late FakeSharedPreferencesAsync store;
+    late SharedPreferencesAsync preferences;
+
+    setUp(() async {
+      store = FakeSharedPreferencesAsync();
+      SharedPreferencesAsyncPlatform.instance = store;
+      preferences =
```

Comment:
See comments elsewhere about potentially using a helper returning a record to avoid the stateful test fixture.